### PR TITLE
Fix serialization of null @nullable fields: omit them entirely.

### PR DESCRIPTION
**Pull request by davidmorgan**
For https://github.com/google/built_json.dart/issues/23

---

This change is [Reviewable](https://reviewable.io/reviews/google/built_json.dart/24)

### CODE REVIEW

File: built_json_generator/test/built_json_generator_test.dart [hunk 1]
```diff
@@ -175,6 +177,91 @@ class _$ValueSerializer implements StructuredSerializer<Value> {
...
+      expect(
+          await generate(r'''
+library value;
+
+import 'package:test_support/test_support.dart';
+
+abstract class Value implements Built<Value, ValueBuilder> {
+  static final Serializer<Value> serializer = _$serializer;
+  bool get aBool;
+  @nullable double get aDouble;
```
Comment (Andersmholmgren) [inline built_json_generator/test/built_json_generator_test.dart]: you may want to consider a second nullable field to make sure the join code works as expected

File: example/lib/value.g.dart [hunk 2]
```diff
@@ -114,19 +129,22 @@ class _$Value extends Value {
...
-    return hashObjects([anInt, aString, anObject, aDefaultInt, listOfInt]);
+    return hashObjects(
+        [anInt, aString, anotherString, anObject, aDefaultInt, listOfInt]);
   }
 
   String toString() {
     return 'Value {'
         'anInt=${anInt.toString()}\n'
         'aString=${aString.toString()}\n'
+        'anotherString=${anotherString.toString()}\n'
```
Comment (Andersmholmgren) [inline example/lib/value.g.dart:147]: should this be

```
anotherString?.toString()
```

anotherString might be null right?

**Comment by Andersmholmgren** [comment]
lgtm. Thanks

File: example/lib/value.g.dart (hunk 2 above)
Comment (davidmorgan) [inline example/lib/value.g.dart:147]: null.toString() returns "null", so this is fine.

**Comment by davidmorgan** [comment]
Thanks, improved the test as suggested. Merging+publishing.

**Pull request merged**